### PR TITLE
Fix `TrainingArguments.label_names` docs to reflect the correct default value behaviour

### LINKED ISSUE #21287: [docs] TrainingArguments default label_names is not what is described in the documentation
### System Info

- `transformers` version: 4.25.1
- Platform: macOS-12.6.1-arm64-arm-64bit
- Python version: 3.8.15
- Huggingface_hub version: 0.11.1
- PyTorch version (GPU?): 1.13.1 (False)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: <fill in>
- Using distributed or parallel set-up in script?: No

### Who can help?

@sgugger, @stevhliu and @MKhalusova

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [X] My own task or dataset (give details below)

### Reproduction

1. Create a model with a `forward` that has more than one label. For example:

```
 def forward(
        self,
        input_ids,
        bbox,
        attention_mask,
        token_type_ids,
        labels,
        reference_labels
)
```

2. Create a trainer for your model with `trainer = Trainer(model, ...)`. Make sure to not set `label_names` and let it default.
3. Check `trainer.label_names` and see that it returns `["labels", "reference_labels"]`

### Expected behavior

[The documentation](https://huggingface.co/docs/transformers/main_classes/trainer#transformers.Seq2SeqTrainingArguments.label_names) states that:

> Will eventually default to ["labels"] except if the model used is one of the XxxForQuestionAnswering in which case it will default to ["start_positions", "end_positions"].

[This PR](https://github.com/huggingface/transformers/pull/16526) changed the behaviour that the documentation describes.

### PR DESCRIPTION
**Pull request by fredtcaroli**
Fixes #21287
@sgugger

### CODE REVIEW

File: src/transformers/training_args.py [hunk 1]
```diff
@@ -352,8 +352,9 @@ class TrainingArguments:
         label_names (`List[str]`, *optional*):
             The list of keys in your dictionary of inputs that correspond to the labels.
 
-            Will eventually default to `["labels"]` except if the model used is one of the `XxxForQuestionAnswering` in
-            which case it will default to `["start_positions", "end_positions"]`.
+            Will eventually default to the list of input keys that contain the word "label", except if the model used is
```
Comment (sgugger) [inline src/transformers/training_args.py:355]:
```suggestion
            Will eventually default to the list of arguments names in `model.forward` that contain the word "label", except if the model used is
```

**Review by sgugger — COMMENTED** [review]
Thanks for adding this! Left a suggestion and make sure to run `make style` on your branch so that the quality checks pass.

File: src/transformers/training_args.py (hunk 1 above)
Comment (fredtcaroli) [reply src/transformers/training_args.py:355]: In that case, should I mention `TF` and `Flax` models that use `call` and `__call__` methods instead of `forward`? Or is this more for internal usage?

Comment (sgugger) [reply src/transformers/training_args.py:355]: Maybe "argument names accepted by the model" then, to be framework-agnostic.

**Comment by fredsensibill** [comment]
This was never flagged as a breaking change (indeed I only found this because it broke one of my scripts). I wonder if I should add "🚨 🚨 🚨" to the PR name to indicate a breaking change

**Review by sgugger — APPROVED** [review]
